Add external refresh metrics

clandry94 opened the pull request:
### What does this PR do?

This adds support for reporting external refresh metrics for the Elasticsearch integration. 

### Motivation

These metrics are newly emitted on ES7.2+. They are provide a better representation of when documents are searchable after being indexed than the base refresh metrics. More details can be found https://github.com/elastic/elasticsearch/pull/38643 and https://github.com/elastic/elasticsearch/issues/36712.

### Additional

Should I make more changes to the metrics.py file? I categorized these within the `PRIMARY_SHARD_METRICS_POST_1_0` and `STATSD_METRICS` maps, but this goes against the comment that they are common to all Elasticsearch versions / all versions post 1.0 since this metric was added in 7.2.0. At the same time, the test file seems to already have a way to deal with metrics added between versions. 

### Review checklist (to be filled by reviewers)

- [ ] PR title must be written as a CHANGELOG entry [(see why)](https://github.com/DataDog/integrations-core/blob/master/CONTRIBUTING.md#pull-request-title)
- [ ] Files changes must correspond to the primary purpose of the PR as described in the title (small unrelated changes should have their own PR)
- [ ] PR must have `changelog/` and `integration/` labels attached
- [ ] Feature or bugfix must have tests
- [ ] Git history [must be clean](https://github.com/DataDog/integrations-core/blob/master/CONTRIBUTING.md#commit-messages)
- [ ] If PR adds a configuration option, it must be added to the configuration file.

FlorianVeaux commented:
Hey @clandry94 
Thanks for making this PR and letting us know about those new metrics. A few comments I have:

1. As you can see our checks are failing, we recently migrated from travis to azure and we did not handle forks correctly. Would you mind rebasing on the latest master so that the tests can run?
2. The new metrics have to be added in the `metadata.csv` file, if you don't mind.
3. Similarly to how we handle version-specific metrics, you should create new constants in the `metrics.py` called `ADDITIONAL_METRICS_POST_7_2_0` and `PRIMARY_SHARD_METRICS_POST_7_2_0` and implement the logic in the methods `stats_for_version` and `pshard_stats_for_version`
4. I tried to run tests locally and there are a few errors popping up because of 3.
5. The `tox.ini` file contains the list of environments on which we run tests, adding an entry for elastic 7.2 would help in the future.

Let us know if we can be of any help :)

PS: Note that you can run tests locally by setting `ddev` and running `ddev test elastic`, [see here](https://github.com/DataDog/integrations-core/tree/master/datadog_checks_dev)

clandry94 commented:
Thanks for the comments @FlorianVeaux. That addresses pretty much every question I had 😄.

clandry94 commented:
Made the changes you requested and CI looks good @FlorianVeaux

FlorianVeaux commented on elastic/tox.ini:
```diff
@@ -46,6 +46,7 @@ setenv =
     6.3: ELASTIC_IMAGE=6.3.2
     ; EOL 2020-02-23
     6.4: ELASTIC_IMAGE=6.4.2
+    7.2: ELASTIC_IMAGE=7.2.0
```
This line sets the `ELASTIC_IMAGE` environment variable when running the tox env 7.2 but this env doesn't yet exist.
You also have to add an entry on line 8 so that tox will run this env.

```
py{27,37}-{5.4,5.5,5.6,6.0,6.1,6.2,6.3,6.4,7.2}
```

FlorianVeaux commented:
Thanks it looks good.
It's not running tests with the 7.2.0 elastic image though, would you mind making the change? Just making sure it works as expected.

clandry94 commented:
Okay, took a bit of extra work but i've:
1) Added ES7.2 to the list of images to test
2) Changed the docker compose for ES tests to use single node discovery type. This will likely fix flaky tests in previous versions of ES and not using single node discovery in test was causing ES7.2 tests to fail 
3) Removed thread_pool.index.* metrics as they were removed in ES7 https://www.elastic.co/guide/en/elasticsearch/reference/7.0/breaking-changes-7.0.html#_index_thread_pool

clandry94 commented:
Rebased and squashed
